app/addons/documents/shared-resources.js

```diff
@@ -36,9 +36,9 @@ define([
     initialize: function (_attrs, options) {
       if (this.collection && this.collection.database) {
         this.database = this.collection.database;
-      } else if (options.database) {
-        this.database = options.database;
+        return;
       }
+      this.database = options.database;
```

Comment:
Seems like the logic changed a little here. The old version only set this value if options.database evaluates to true; this always sets it.